### ISSUE TITLE
Suggestion: Custom labels for tabs

### ISSUE THREAD

**Issue by MarekJuracka** [mbnuqw/sidebery#106]
## Description
Would it be possible to assign custom label to each tab? Right now the tab shows website title, it would be great if you could assign a colored label with custom text for each tab right after favicon before the title text

## Use cases
For me personally, i have multiple instances of same webapplication open for various environments.
Unfortnunately they share the same favicon and same title so they are hard to distinguish without opening them, combined with tab unloading, you have to wait few seconds before seeing you opened a wrong one. I tried using groups for this and it kind of works for small number of tabs. Once you grow to 83 for example as i'm now it's hard to navigate.

**Comment by mbnuqw** [comment]
Nice idea! I think I can add tags feature for the tabs.

Just some experiments:
1. Simple colored badge
![screenshot-2019-11-19 22:49:12](https://user-images.githubusercontent.com/6276694/69172249-5075a380-0b1f-11ea-886e-037c8132bfea.png)

2. One inline tag (as you described)
![screenshot-2019-11-19 22:47:01](https://user-images.githubusercontent.com/6276694/69172301-65523700-0b1f-11ea-8616-a163bf783b1b.png)

3. Multiple tags on the second line
![screenshot-2019-11-19 22:36:15](https://user-images.githubusercontent.com/6276694/69172400-8fa3f480-0b1f-11ea-9b06-d2c13ba9c6f7.png)

**Comment by mbnuqw** [comment]
Also, you can try to use [Tab ReTitle](https://addons.mozilla.org/en-US/firefox/addon/tab-retitle/?src=search) to change titles.

**Comment by n4st3** [comment]
Yea the second option looks great, maybe even with multiple tags?

**Comment by gantheaume** [comment]
Totally agree, I would be very interested in a manual badge/tag/flag toggle as shown in the experiments

[Tab ReTitle](https://addons.mozilla.org/en-US/firefox/addon/tab-retitle/?src=search) doesn't seem to have a simple "manual mode", it's all about automation I'm not interested about. I would just like a flag I set on/off by hand, mainly for read/unread/"to come back to" usage. Maybe even being able to set/unset the "native" unread corner flag would suit me.

I didn't find any other extension that does the job.

**Comment by Iss-in** [comment]
yeah, this would be awesome. Edge browser does it really well imo. If nothing, we could have it as template

**Comment by RokeJulianLockhart** [comment]
> https://github.com/mbnuqw/sidebery/issues/106#issue-525063786

If of interest to anyone who merely needs this functionality *somehow* in Firefox, rather than specifically via Sideberry, I've filed [`bugzilla.mozilla.org/show_bug.cgi?id=1958470#c0`](https://bugzilla.mozilla.org/show_bug.cgi?id=1958470#c0:~:text=Implement%20non%2Dhierarchical%20labels%20to%20organise%20tabs%20with.) for its native vertical tab implementation.

**Comment by the-nelsonator** [comment]
This is covered by the Edit Title option in right-click menu (at least the original request)

**Comment by RokeJulianLockhart** [comment]
> https://github.com/mbnuqw/sidebery/issues/106#issuecomment-2785095765

@the-nelsonator, indeed, albeit solely ½ of the description section. I've split them undermentioned to demonstrate:

> Right now the tab shows website title, it would be great if you could either:
> 
> 1. Set the website title to whatever custom text, or
> 2. Assign a colored label with custom text for each tab right after favicon before the title text.

@MarekJuracka, probably best to remove the first point, so that the scope is clear.

**Comment by anthmapper** [comment]
I personally would like something I can toggle with a keyboard shortcut. It's for archiving purposes.

My use case:
- ctrl+tab progressively through all my open tabs
- on the ones I want, toggle the flag (ideally via the configurable shortcut I would probably set to ctrl+LETTER or something)
- Right-click -> copy toggled tabs to clipboard (ideally with tree structure)
ALTERNATIVELY
- Right-click -> close all [toggled or untoggled] tabs.

Option 1. [in @mbnuqw 's message above](https://github.com/mbnuqw/sidebery/issues/106#issuecomment-555643495) would be plenty enough

>     1. Simple colored badge
![screenshot-2019-11-19 22:49:12](https://user-images.githubusercontent.com/6276694/69172249-5075a380-0b1f-11ea-886e-037c8132bfea.png)


No need to keep a memory of the flagged tabs past the current session in my case.
As I stated above, re-enabling the little "unread corner" would be sufficient too (providing I can then close all [read or unread], or copy to clipboard all [read or unread] in one move)

**Comment by mbnuqw** [comment]
@anthmapper, created FR: #2097

**Comment by mbnuqw** [comment]
Closing this as done (Editing title of the tab). For tags: #2102
